crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -317,6 +364,197 @@ impl IndexeddbCryptoStore {
+                    value: self.serializer.serialize_value(&pickle)?,
+                });
+            }
+        }
+
+        let device_changes = &changes.devices;
+        let identity_changes = &changes.identities;
+        let olm_hashes = &changes.message_hashes;
+        let key_requests = &changes.key_requests;
+        let withheld_session_info = &changes.withheld_session_info;
+        let room_settings_changes = &changes.room_settings;
+
+        let device_store = indexeddb_changes.get(keys::DEVICES);
+
+        if !device_changes.new.is_empty() || !device_changes.changed.is_empty() {
+            for device in device_changes.new.iter().chain(&device_changes.changed) {
+                let key = self
+                    .serializer
+                    .encode_key(keys::DEVICES, (device.user_id(), device.device_id()));
+                let device = self.serializer.serialize_value(&device)?;
+
+                device_store.push(DbOperation::PutKeyVal { key, value: device });
+            }
+        }
+
+        if !device_changes.deleted.is_empty() {
+            for device in &device_changes.deleted {
+                let key = self
+                    .serializer
+                    .encode_key(keys::DEVICES, (device.user_id(), device.device_id()));
+                device_store.push(DbOperation::Delete(key));
+            }
+        }
+
+        if !identity_changes.changed.is_empty() || !identity_changes.new.is_empty() {
+            let identities = indexeddb_changes.get(keys::IDENTITIES);
+            for identity in identity_changes.changed.iter().chain(&identity_changes.new) {
+                identities.push(DbOperation::PutKeyVal {
+                    key: self.serializer.encode_key(keys::IDENTITIES, identity.user_id()),
+                    value: self.serializer.serialize_value(&identity)?,
+                });
+            }
+        }
+
+        if !olm_hashes.is_empty() {
+            let hashes = indexeddb_changes.get(keys::OLM_HASHES);
+            for hash in olm_hashes {
+                hashes.push(DbOperation::PutKeyVal {
+                    key: self
+                        .serializer
+                        .encode_key(keys::OLM_HASHES, (&hash.sender_key, &hash.hash)),
+                    value: JsValue::TRUE,
+                });
+            }
+        }
+
+        if !key_requests.is_empty() {
+            let gossip_requests = indexeddb_changes.get(keys::GOSSIP_REQUESTS);
+
+            for gossip_request in key_requests {
+                let key_request_id = self
+                    .serializer
+                    .encode_key(keys::GOSSIP_REQUESTS, gossip_request.request_id.as_str());
+                let key_request_value = self.serialize_gossip_request(gossip_request)?;
+                gossip_requests
+                    .push(DbOperation::PutKeyVal { key: key_request_id, value: key_request_value });
```

Comment:
Note for the reviewer: This used to be a `put_key_val_owned` but I don't think it was needed. Let me know

Reply:
That seems correct to me, good catch :+1: